dask/dataframe/io/parquet/arrow.py

```diff
@@ -685,9 +390,28 @@ def read_metadata(
             elif not gather_statistics:
                 gather_statistics = True
```

Comment:
A comment on the lines above (but cannot comment on it because github hides it by default). 

The logic of raising an error if `gather_statistics is False` when there are filters might not be correct anymore for the new dataset engine (since gather statistics is now mainly used for the divisions? and pyarrow will always check the statistics when a filter is passed)

Reply:
That sounds right - I'll move this into `ArrowLegacyEngine._gather_metadata`, since it is not true for `ArrowDatasetEngine`.